### PR TITLE
Wayland: start gnome-session-x11-services-ready.target

### PR DESCRIPTION
**Pull request by lenemter**
Fixes https://github.com/elementary/gala/issues/1967
Fixes https://github.com/elementary/gala/issues/2113
Fixes https://github.com/elementary/gala/issues/2131

### CODE REVIEW

**Review by leolost2605 — CHANGES_REQUESTED** [review]
Only some minor comments apart from that makes sense to me but I don't really know the inner workings here :)

**Review by vjr — COMMENTED** [review]
I can confirm it fixes at least #1967 for me!

**Review by vjr — APPROVED** [review]

**Comment by vjr** [comment]
@lenemter just wondering whether instead of doing this in code is it not possible/advisable to configure this systemd service/unit to start via the config files?

**Comment by lenemter** [comment]
@vjr gnome shell does it this way, I think they know how to do this better than me :)

File: src/WindowManager.vala [hunk 1]
```diff
@@ -148,7 +148,45 @@ namespace Gala {
...
+                });
+            }
+#endif
+        }
+
+#if WITH_SYSTEMD
+        private async void start_x11_services (GLib.Task task) {
+            try {
+                var session_bus = GLib.Bus.get_sync (GLib.BusType.SESSION);
+                session_bus.call.begin (
```
[the remark this answers is not in the document]
Comment (lenemter) [reply src/WindowManager.vala]: Done!

File: src/WindowManager.vala [hunk 2]
```diff
@@ -148,7 +148,45 @@ namespace Gala {
...
+                    return true;
+                });
+            }
+#endif
+        }
+
+#if WITH_SYSTEMD
+        private async void start_x11_services (GLib.Task task) {
+            try {
+                var session_bus = GLib.Bus.get_sync (GLib.BusType.SESSION);
```
[the remark this answers is not in the document]
Comment (lenemter) [reply src/WindowManager.vala]: Fixed

**Pull request merged**